app/controllers/trips_controller.rb

```diff
@@ -1,10 +1,19 @@
 class TripsController < ApplicationController
   before_action :set_trip, only: %i[ show edit update destroy ]
-  before_action :authenticate_user!
+  before_action :only => [:edit] do
+    redirect_to new_user_session_path unless current_user && current_user.role == "admin"
+  end
 
   # GET /trips or /trips.json
   def index
-    @trips = Trip.all
+    query = Trip
```

Comment:
I want to see tests for this, at least 7 (one for each param plus one without any)

Reply:
I added this in the table card #30